opentelemetry-sdk/src/runtime.rs

```diff
@@ -131,3 +132,108 @@ impl Runtime for AsyncStd {
         Box::pin(async_std::task::sleep(duration))
     }
 }
+
+/// `MessageRuntime` is an extension to [`Runtime`]. Currently, it provides a
+/// channel that is used by the [log] and [span] batch processors.
+///
+/// [log]: crate::logs::BatchLogProcessor
+/// [span]: crate::trace::BatchSpanProcessor
+pub trait MessageRuntime<T: Debug + Send>: Runtime {
```

Comment:
I hadn't really noticed this before (and I note the name was equally confusing before), but I think the `*Runtime` name for this is actually pretty bad. How about naming it `RuntimeChannel`? Given the implementations for `tokio::sync::mpsc` and `async_std::channel` that seems like a much more descriptive name.

Reply:
Agreed, `Message` doesn't communicate that much. That being said, `MessageRuntime` seems to be only used for the batch processor, so what about something like `BatchRuntime`?

However, if there are plans to use this trait for purposes other than batch processing, why not just rename it to `RuntimeExt`?

Reply:
These types aren't runtimes, they're channels. Calling it runtime is confusing at best.

Reply:
> However, if there are plans to use this trait for purposes other than batch processing, why not just rename it to RuntimeExt?

It would be nice if batch processors could have a way to spawn additional async work and use the configured runtime to drive it to completion as part of their export. I imagine this could look like a `BatchProcessorRuntimeExt`  (or perhaps just an updated `Runtime` trait itself).

```rust
pub trait Runtime: Clone + Send + Sync + 'static {
<snip>
    /// Spawn a new task or thread, which executes the given future.
    ///
    /// # Note
    ///
    /// This is mainly used to run batch span processing in the background. Note, that the function
    /// does not return a handle. OpenTelemetry will use a different way to wait for the future to
    /// finish when TracerProvider gets shutdown. At the moment this happens by blocking the
    /// current thread. This means runtime implementations need to make sure they can still execute
    /// the given future even if the main thread is blocked.
    fn spawn(&self, future: BoxFuture<'static, ()>);
```

Reply:
Making such a change is certainly out of scope of this PR. But maybe ideas for the future can help pick a good name right now.